backend-api/src/functions/asyncBiometricToken/validateRequestBody/validateRequestBody.test.ts

```diff
@@ -0,0 +1,79 @@
+import { validateRequestBody } from "./validateRequestBody";
+
+describe("Validate request body", () => {
+  describe.each([
+    [
+      "Given body is null or undefined",
+      null,
+      "Request body is either null or undefined. Request body: null",
+    ],
+    [
+      "Given body cannot be parsed as JSON",
+      "foo",
+      `Request body could not be parsed as JSON. SyntaxError: Unexpected token 'o', "foo" is not valid JSON`,
+    ],
+  ])("%s", (_description, body, errorMessage) => {
+    it("Returns an error result", () => {
+      const result = validateRequestBody(body);
+
+      expect(result).toStrictEqual({
+        isError: true,
+        value: {
+          errorMessage,
+        },
+      });
+    });
+  });
+
+  describe.each([
+    [
+      "Given sessionId is not present",
+      {},
+      "sessionId in request body is either null or undefined. sessionId: undefined",
+    ],
+    [
+      "Given sessionId is not a string",
+      { sessionId: 123 },
+      "sessionId in request body is not of type string. sessionId: 123",
+    ],
+    [
+      "Given sessionId is an empty string",
+      { sessionId: "" },
+      "sessionId in request body is an empty string. sessionId: ",
+    ],
+    [
+      "Given documentType is not present",
+      { sessionId: "58f4281d-d988-49ce-9586-6ef70a2be0b4" },
+      "documentType in request body is either null or undefined. documentType: undefined",
+    ],
+    [
+      "Given documentType is not a string",
+      { sessionId: "58f4281d-d988-49ce-9586-6ef70a2be0b4", documentType: 123 },
+      "documentType in request body is not of type string. documentType: 123",
+    ],
+    [
+      "Given documentType is an empty string",
+      { sessionId: "58f4281d-d988-49ce-9586-6ef70a2be0b4", documentType: "" },
+      "documentType in request body is an empty string. documentType: ",
+    ],
+    [
+      "Given documentType is an invalid document type",
+      {
+        sessionId: "58f4281d-d988-49ce-9586-6ef70a2be0b4",
+        documentType: "BUS_PASS",
+      },
+      "documentType in request body is invalid. documentType: BUS_PASS",
+    ],
+  ])("%s", (_description, body, errorMessage) => {
+    it("Returns an error result", () => {
+      const result = validateRequestBody(JSON.stringify(body));
```

Comment:
This is why I have two describe.each FYI - I didn't want to include JSON.stringify in the params above as it would take up too much space so I do it directly in the request once. This will however fail for the 2 tests in the first describe.each block